helper/restore_helper.go

```diff
@@ -351,6 +378,10 @@ func doRestoreAgentInternal(restoreHelper IRestoreHelper) error {
 						logWarn(fmt.Sprintf("Oid %d, Batch %d: Skip file discovered, skipping this relation.", tableOid, batchNum))
 						err = nil
 						skipOid = tableOid
+						if *singleDataFile && readers[contentToRestore] != nil && readers[contentToRestore].getReaderType() == SUBSET {
+							bytesToDiscard := int64(end[contentToRestore] - start[contentToRestore])
+							_, err = readers[contentToRestore].discardData(bytesToDiscard)
```

Comment:
How good is it that you erased the nil error value?

Reply:
Really?! After moving to `LoopEnd`, we get into the condition `if err != nil {` and without `onErrorContinue` we return from the function! NOT continue!

Reply:
There is no hanging without `onErrorContinue`. I wrote about the case when `onErrorContinue` is set

Reply:
Before the patch, without the `onErrorContinue` option, there was a continuation of the loop, and after the patch, there was a return from the function.